src/value_generators/test/DocGenYamlTestCaseRunner.cpp

```diff
@@ -83,7 +83,8 @@ class ValGenTestCase {
             for (const auto&& nextValue : this->_thenReturns) {
                 auto expected = testing::toDocumentBson(nextValue);
                 auto actual = docGen();
-                REQUIRE(toString(expected.view()) == toString(actual.view()));
+                auto expectedFix = bsoncxx::from_json(toString(expected.view()));
+                REQUIRE(toString(expectedFix.view()) == toString(actual.view()));
```

Comment:
thanks for making this fix and working with drivers to uncover the root of this issue. Can you add a TODO comment with a link to the TIG/PERF ticket to remove once the bson bug is fixed?

Reply:
Nice!

Reply:
Wait, does this work? I thought you would need the extra round trip on the "actual" value, not the expected value.